### PR TITLE
Implement IsVisible property in ViewHandlers

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Implement `IsVisible` property in ViewHandlers.

### Platforms Affected ### 

- Core
- iOS
- macOS
- Android
- Windows

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [x] Targets a single property for a single control (or intertwined few properties)
- [x] Adds the property to the appropriate interface
- [x] Avoids any changes not essential to the handler property
- [x] Adds the mapping to the PropertyMapper in the handler
- [x] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [x] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [x] Tags ported renderer methods with [PortHandler]
- [x] Adds an example of the property to the sample project (MainPage)
- [x] Adds the property to the stub class
- [x] Implements basic property tests in DeviceTests

### CODE REVIEW

**Comment by GalaxiaGuy** [comment]
Is there any hope of expanded visibility functionality as outlined here: https://github.com/dotnet/maui/discussions/33

File: src/Core/src/Platform/Android/ViewExtensions.cs [hunk 1]
```diff
@@ -14,6 +15,17 @@ public static void UpdateIsEnabled(this AView nativeView, IView view)
...
 
+		public static void UpdateIsVisible(this AView nativeView, IView view)
+		{
+			if (view.IsVisible && nativeView.Visibility != ViewStates.Visible)
+				nativeView.Visibility = ViewStates.Visible;
+			if (!view.IsVisible && nativeView.Visibility != ViewStates.Gone)
+				nativeView.Visibility = ViewStates.Gone;
+
+			nativeView.Invalidate();
+			nativeView.RequestLayout();
```
Comment (mattleibow) [inline src/Core/src/Platform/Android/ViewExtensions.cs:26]: Is this the correct set of operations?

Also, we have all the checks above, but then still just fire a re-layout. We probably should only do this when it actually changed.

**Review by mattleibow — COMMENTED** [review]
Looks safe, except that Android is the only platform that has some "request layout" code. Is this normal? Are the other platforms just that smart and Android not?

**Comment by hartez** [comment]
> Is there any hope of expanded visibility functionality as outlined here: #33

We're pointing folks at that discussion internally. I don't know how that will turn out, but even if we decide on the expanded visibility, we _might_ merge this PR and then add in the expanded stuff later. 

So what I'm saying is, "don't take the merging of this PR to mean that the expanded version won't happen".

**Comment by hartez** [comment]
Closing in favor of #951

**Pull request closed without merging**